### PR TITLE
Feature drs

### PR DESCRIPTION
**Pull request by snmpboy**
### Description

Added drs folder in libcloud/ for disaster recovery driver. NTTC-CIS provides a disaster recovery service for VM's.

### Status

Replace this: describe the PR status. Examples:

- done, ready for review

### Checklist (tick everything that applies)

- [x] [Code linting](http://libcloud.readthedocs.org/en/latest/development.html#code-style-guide) (required, can be done after the PR checks)
- [x] Documentation
- [x] [Tests](http://libcloud.readthedocs.org/en/latest/testing.html)
- [x] [ICLA](http://libcloud.readthedocs.org/en/latest/development.html#contributing-bigger-changes) (required for bigger changes)

### CODE REVIEW

**Comment by snmpboy** [comment]
See JIRA libcloud1023.

File: libcloud/test/compute/test_nttcis.py [hunk 1]
```diff
@@ -73,29 +74,29 @@ def test_node_extras(driver):
     assert isinstance(ret[0].extra['cpu'], NttCisServerCpuSpecification)
     assert isinstance(ret[0].extra['disks'], list)
     assert isinstance(ret[0].extra['disks'][0], NttCisServerDisk)
-    assert ret[0].extra['disks'][0].size_gb == 50
+    assert ret[0].extra['disks'][0].size_gb, 10
```
Comment (tonybaloney) [inline libcloud/test/compute/test_nttcis.py:77]: the comma will be the error message, not the asserted value . So this test just passes regardless

File: libcloud/test/compute/test_nttcis.py [hunk 2]
```diff
@@ -73,29 +74,29 @@ def test_node_extras(driver):
...
     assert isinstance(ret[1].extra['disks'], list)
     assert isinstance(ret[1].extra['disks'][0], NttCisServerDisk)
-    assert ret[1].extra['disks'][0].size_gb == 50
+    assert ret[1].extra['disks'][0].size_gb, 10
```
Comment (tonybaloney) [inline libcloud/test/compute/test_nttcis.py:80]: as above

File: libcloud/drs/base.py [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+class DRSConsistencyGroup(object):
+    """
+    Provide a common interface for handling DRS.
+    """
+
+    def __init__(self, id, name, description, journalSizeGB,
+                 serverPairSourceServerId, serverPairtargetServerId,
+                 driver, extra=None):
+        """
+        :param id: Load balancer ID.
```
Comment (tonybaloney) [inline libcloud/drs/base.py:35]: assume this is copy+pasta from loadbalancer module. Please correct the docstrings

File: libcloud/drs/base.py [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+
+    def __init__(self, key, secret=None, secure=True, host=None,
+                 port=None, **kwargs):
+        super(DRSDriver, self).__init__(key=key, secret=secret, secure=secure,
+                                        host=host, port=port, **kwargs)
+
+    def create_consistency_group(self, name, journal_sz_gb,
+                                 source_server_id, target_server_id):
+        """
+        :param name: Name of the consistency group to create
```
Comment (tonybaloney) [inline libcloud/drs/base.py:87]: Follow style guidelines for the docstrings

File: libcloud/drs/base.py [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+            'delete_consistency_group not implemented for this driver')
+
+    def list_consistency_group_snapshots(self, consistency_group_id):
+        """
+        Return a list of consistency group snapshots.
+
+        :param consistency_group_id: id of a consistency group to fetch
+                                     snapshots from.
+        :type  consistency_group_id: ``str``
+        :rtype: ``list``
```
Comment (tonybaloney) [inline libcloud/drs/base.py:145]: List of what type?

File: libcloud/drs/base.py [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+
+        raise NotImplementedError(
+            'list_consistency_group_snapshots not implemented for this driver')
+
+    def expand_journal(self, consistency_group_id, size_gb):
+        """
+        :param consistency_group_id: consistency group's id with journal
+                                     to expand
+        :type consistency_group_id: ``str``
+        :param size_gb: Size in increments of 10 Gb to expand journal.
```
Comment (tonybaloney) [inline libcloud/drs/base.py:156]: pls follow style guidelines for docstrings

File: libcloud/drs/base.py [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+        """
+
+        raise NotImplementedError(
+            'expand_journal not implemented for this driver')
+
+    def start_failover_preview(self, consistency_group_id, snapshot_id):
+        """
+        :param consistency_group_id: consistency group's id with journal
+                                     to expand
+        :type consistency_group_id: ``str ``
```
Comment (tonybaloney) [inline libcloud/drs/base.py:168]: as above

File: libcloud/drs/base.py [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+        """
+
+        raise NotImplementedError(
+            'stop_failover_preview not implemented for this driver')
+
+    def initiate_failover(self, consistency_group_id):
+        """
+        :param consistency_group_id: Consistency group id of consistency
+                                     group on which to initiate failover.
+        :type consistency_group_id: ``str``
```
Comment (tonybaloney) [inline libcloud/drs/base.py:195]: as above

File: libcloud/common/nttcis.py [hunk 9]
```diff
@@ -1924,3 +1926,259 @@ def __repr__(self):
         return ('<NttCisNic: private_ip_v4=%s, vlan=%s,'
                 'network_adapter_name=%s>'
                 % (self.private_ip_v4, self.vlan, self.network_adapter_name))
+
```
Comment (tonybaloney) [inline libcloud/common/nttcis.py:1929]: These changes seem unrelated to the DRS implementation?

**Comment by tonybaloney** [comment]
Thanks @snmpboy the implementation of a new base driver type looks good.

Typically we look for at least 3 drivers in a new base driver type before it gets merged into the project. This is to ensure the API is going to map to multiple platforms. 

I'd want to see example implementations for at least AWS, Azure and GCP

See this example on the container modules (written before Kubernetes became the de-facto standard anyway)
http://libcloud.apache.org/blog/2016/02/05/libcloud-containers-example.html

**Comment by tonybaloney** [comment]
@Kami @pquentin request for comments please
